rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+such as regulatory compliance reporting, where specific transaction details must be revealed to meet legal requirements, while other sensitive transactions remain private.
+
+#### a) Stages of Deshielded Executions
+
+* Stage 1 (UTXO Consumption): The process begins in the private state, where UTXOs are consumed as inputs for the transaction. This involves gathering all necessary 
+UTXOs that contain the assets or balances to be made public, as well as any associated private data stored in memory slots.
+
+* Stage 2 (Public State Modification): After the UTXOs are consumed, the transaction details are made public by modifying the public state. This update includes changes 
+to the public balances, storage data, and any necessary public records. While the public state is updated, the sender’s identity and other sensitive information remain hidden, 
+thanks to the privacy-preserving properties of ZKPs.
+
+This model ensures that private data can be selectively revealed when needed, offering both flexibility and transparency. It is particularly useful for scenarios requiring 
+auditing or compliance reporting, where specific details must be made publicly verifiable without exposing the entire history or contents of private transactions.
+
+#### b) How Deshielded Executions Work
+The deshielded execution process starts when a user initiates a transaction using private UTXOs. The Nescience zkVM is called to generate a zero-knowledge proof, 
+which validates the transaction without revealing sensitive details such as the sender's identity or the specifics of the Nescience application being executed.
+
+During the transaction, the UTXOs from the private state are consumed, meaning they are used up as inputs and will no longer be available for future transactions. 
+Instead of generating new UTXOs, the transaction modifies the public state, updating the necessary balances or memory slots related to the transaction. 
+Here’s a step-by-step breakdown of how the deshielded execution process unfolds:
+
+1. Get Receiver's Public Address: The sender first identifies the public address of the receiver, to which the information or assets will be made public.
+
+2. Determine Input UTXOs and Public State Modifications: The sender gathers all the input UTXOs needed for the transaction and determines the public state modifications 
+necessary for the Nescience applications and token transfers involved.
+
+3. Calculate Nullifiers: Nullifiers are generated for each input UTXO, ensuring that these UTXOs cannot be reused or double-spent. The nullifiers are derived from the
+ corresponding UTXO commitments.
+
+4. Call zkVM with Deshielded Circuits: The sender invokes the zkVM with deshielded kernel circuits, which generates the proof. The proof ensures that all input UTXOs 
+are valid by verifying their membership in the UTXO tree and their non-membership in the nullifier tree, ensuring they haven’t been spent.
```

Comment:
we could make this sentence more elegant but rephrasing to something like this:
"The proof ensures that all input UTXOs are valid and haven't previously been spent by verifying their membership in the UTXO tree and their non-membership in the nullifier tree."